src/app.module.ts

```diff
@@ -18,37 +18,50 @@ import { DepositModule } from "./modules/deposit/module";
 import { AuthModule } from "./modules/auth/auth.module";
 import { UserModule } from "./modules/user/module";
 import { AirdropModule } from "./modules/airdrop/module";
+import { ModuleOptions, RunMode } from "./dynamic-module";
 
-@Module({
-  imports: [
-    ConfigModule.forRoot({
-      ignoreEnvFile: false,
-      ignoreEnvVars: false,
-      isGlobal: true,
-      expandVariables: true,
-      load: [configuration],
-    }),
-    TypeOrmModule.forRootAsync({
-      imports: [DatabaseModule],
-      useExisting: TypeOrmDefaultConfigService,
-    }),
-    HealthModule,
-    Web3Module,
-    ScraperModule,
-    BullModule.forRootAsync({
-      imports: [MessagingModule],
-      useExisting: BullConfigService,
-    }),
-    ReferralModule,
-    MarketPriceModule,
-    ScheduleModule.forRoot(),
-    DepositModule,
-    AuthModule,
-    UserModule,
-    AirdropModule,
-  ],
-})
+@Module({})
 export class AppModule {
+  static forRoot(moduleOptions: ModuleOptions): DynamicModule {
+    const imports = [
+      ConfigModule.forRoot({
+        ignoreEnvFile: false,
+        ignoreEnvVars: false,
+        isGlobal: true,
+        expandVariables: true,
+        load: [configuration],
+      }),
+      TypeOrmModule.forRootAsync({
+        imports: [DatabaseModule],
+        useExisting: TypeOrmDefaultConfigService,
+      }),
+      HealthModule,
+      Web3Module,
+      ReferralModule.forRoot(moduleOptions),
+      MarketPriceModule,
+      ScheduleModule.forRoot(),
+      DepositModule,
+      AuthModule.forRoot(moduleOptions),
+      UserModule.forRoot(moduleOptions),
+      AirdropModule.forRoot(moduleOptions),
+    ];
+
+    if ([RunMode.Scraper, RunMode.Test].includes(moduleOptions.runMode)) {
+      imports.push(
+        ScraperModule.forRoot(moduleOptions),
+        BullModule.forRootAsync({
+          imports: [MessagingModule],
+          useExisting: BullConfigService,
+        }),
+      );
+    }
```

Comment:
Are these the only modules that should not be included if running in `normal` mode?